core/io/resource.cpp

```diff
@@ -60,32 +60,32 @@ void Resource::set_path(const String &p_path, bool p_take_over) {
 		p_take_over = false; // Can't take over an empty path
 	}
 
-	ResourceCache::lock.lock();
+	{
+		MutexLock lock(ResourceCache::lock);
```

Comment:
We should probably rename `ResourceCache::lock` to `ResourceCache::mutex` at some point as it's just a recipe for confusion having an incorrect name for it, left it alone in this PR though